driver/pelican/types/thermSchedule.go

```diff
@@ -0,0 +1,246 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/parnurzeal/gorequest"
+	rrule "github.com/teambition/rrule-go"
+)
+
+// Login, Authentication, Thermostat ID Retrieval Structs
+type thermIDRequest struct {
+	Resources []thermIDResources `json:"resources"`
+}
+
+type thermIDResources struct {
+	Children    []thermIDChild `json:"children"`
+	GroupId     string         `json:"groupId"`
```

Comment:
It seems like some of the fields in the structs are unnecessary -- I never see them being used. This includes `GroupId` here and the `Permissions` fields in a couple places.

I don't think you have to include every field present in the source JSON as a field in your struct representation. Fields in the JSON that don't have a matching field in the target struct will be ignored.

Also, we talked about the idea of unmarshalling into a map in places to avoid defining a bunch of structs. Did you try doing that?

Reply:
Noted, I will make sure to eliminate any fields that are not needed.

I will try to using a map to unmarshal. I've given this a shot thus far. Although it saves me the structs, I have to replace the unraveling process with a bunch of "map[string]interface{}" and "[]interface{}" type assertions which makes the code look very messy.

Reply:
Quick update, I've reimplemented the JSON decoding code with free form / empty interfaces and casting. I've removed all the intermediary structs from before. Personally, I think the resulting code is more robust, but the slight con, I feel, is that the JSON decoding (especially all the type assertions) can be a little confusing. Nevertheless, the functionality is still intact and works as expected.